action-dynamic.Rmd

```diff
@@ -654,7 +654,7 @@ server <- function(input, output, session) {
   })
   
   output$plot <- renderPlot({
-    cols <- map_chr(col_names(), ~ input[[.x]])
+    cols <- map_chr(col_names(), ~ input[[.x]] %||% "")
```

Comment:
I think it would be better to use `NA` here, so you could skip the next line.

Reply:
And yeah, you can drop from the `textInput()` generation since a `NULL` is an ok value for `value`.

Reply:
Replacing the ` %||% ""` with `%||% NA` in the `map_chr()` anonymous function results in `Warning: Error in rect: invalid color name ''` if you drop the `cols[cols == ""] <- NA` line, and the palette plot will only show up when you've entered the correct number of (valid) color names.

Both ` %||% NA` or `%||%""` do, however, get rid of the `Warning: Error in : Result 1 must be a single string, not NULL of length 0`. They both just need to handle the `""` to `NA` conversion. 

If going the tidyverse route, something like

`cols <- map_chr(col_names(), ~ input[[.x]] %||% NA) %>% na_if("")`

would work, but I'm not convinced this is better, for the sake of clarity, than your example with `cols[cols == ""] <- NA` on its own line.

Reply:
Having look at this in more detail, I think a better approach is to construct a `default_val()` function specifically for shiny and use that. Do you mind reverting your changes here, and then I'll fix locally.

Reply:
Reverted the code changes, but kept the prose typos I'd found